src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -193,7 +193,14 @@ def preprocess(self, inputs, chunk_length_s=0, stride_length_s=None):
         extra = {}
         if isinstance(inputs, dict):
             stride = inputs.pop("stride", None)
-            _inputs = inputs.pop("raw")
+            # Accepting `"array"` which is the key defined in `datasets` for
+            # better integration
+            if not ("sampling_rate" in inputs and ("raw" in inputs or "array" in inputs)):
+                raise ValueError(
+                    'When passing a dictionnary to AutomaticSpeechRecognitionPipeline, the dict needs to contain a "raw" key containing the numpy array representing the audio and a "sampling_rate" key, containing the sampling_rate associated with that array'
```

Comment:
```suggestion
                    'When passing a dictionnary to AutomaticSpeechRecognitionPipeline, the dict needs to contain a '
                    '"raw" key containing the numpy array representing the audio and a "sampling_rate" key, '
                    "containing the sampling_rate associated with that array"
```
Let's split this on several lines to respect our 119 char limit :-)